src/csvcubed/utils/validations.py

```diff
@@ -91,19 +97,222 @@ def validate_int_type(
     """
     This function will validate if the Optional argument provided is a None value it will return an empty list,
     else it returns any errors returned by the item validation function .
     """
 
     def _validate(
         maybe_item: Optional[T], property_name: str
-    ) -> List[ValidateModelProperiesError]:
+    ) -> List[ValidateModelPropertiesError]:
         if maybe_item is None:
             return []
 
         return validate_item(maybe_item, property_name)
 
     return _validate
+
+
+def validate_float_type(
+    value: float, property_name: str
+) -> List[ValidateModelPropertiesError]:
+    """
+    This function will validate if the argument provided is in fact a float type and,
+    returns any errors returned by the item validation function.
+    """
+    if not isinstance(value, float):
+        return [
+            ValidateModelPropertiesError(
+                "This variable should be a float value, check the following variable:",
+                property_name,
+            )
+        ]
+    elif isnan(value):
+        return [
+            ValidateModelPropertiesError(
+                "This variable should be a float value but is Not a Number (NaN), check the following variable:",
+                property_name,
+            )
+        ]
+    elif isinf(value):
+        return [
+            ValidateModelPropertiesError(
+                "This variable should be a float value but is +-infinity, check the following variable:",
+                property_name,
+            )
+        ]
+
+    return []
+
+
+def validate_file(
+    value: Path, property_name: str
+) -> List[ValidateModelPropertiesError]:
+    if isinstance(value, Path):
+        """
+        This function validates whether the given argument is in fact of type Path, and if not,
+        returns any validation errors raised by the validation function.
+        """
+        if not value.exists():
+            return [
+                ValidateModelPropertiesError(
+                    "This file does not exist, check the following variable:",
+                    property_name,
+                )
+            ]
+        else:
+            return []
+    else:
+        return [
+            ValidateModelPropertiesError(
+                "This is not a valid file path, check the following variable:",
+                property_name,
+            )
+        ]
+
+
+def date(value: dt.date, property_name: str) -> List[ValidateModelPropertiesError]:
+    """
+    This function will validate if the argument provided is in fact a date type and,
+    returns any errors returned by the item validation function.
+    """
+    if not isinstance(value, dt.date):
+        return [
+            ValidateModelPropertiesError(
+                "This is not a valid date format, check the following variable:",
+                property_name,
+            )
+        ]
+    return []
+
+
+def datetime(
```

Comment:
Can we get rid of this and the date function now that we have a function which checks if something is an instance of a class>?

Reply:
Removed them and replaced anywhere they were used with is_instance_of